python/lsst/sims/photUtils/Sed.py

```diff
@@ -636,15 +654,32 @@ def multiplySED(self, other_sed, wavelen_step=PhotometricDefaults.wavelenstep):
     ## routines related to magnitudes and fluxes
 
     def calcADU(self, bandpass, wavelen=None, fnu=None,
-                expTime=PhotometricDefaults.exptime, effarea=PhotometricDefaults.effarea,
-                gain=PhotometricDefaults.gain):
+                photParams=None):
         """
         Calculate the number of adu from camera, using sb and fnu.
 
         Given wavelen/fnu arrays or use self. Self or passed wavelen/fnu arrays will be unchanged.
         Calculating the AB mag requires the wavelen/fnu pair to be on the same grid as bandpass;
          (temporary values of these are used).
+
+        @param [in] bandpass is an instantiation of the Bandpass class
+
+        @param [in] wavelen (optional) is the wavelength grid in nm
+
+        @param [in] fnu (optional) is the flux in Janskys
+
+        If wavelen and fnu are not specified, this will just use self.wavelen and
+        self.fnu
+
+        @param [in] photParams is an instantiation of the
+        PhotometricParameters class that carries details about the
+        photometric response of the telescope.
         """
+
+        if photParams is None:
```

Comment:
Makes sense.  I was enamored of the idea of not having to remember the order of arguments whenever I call these functions, but I was forgetting that python allows you to assign args using the fn(x=1, y=2, z=3) formalism, as well.

I have 'demoted' photParams from a kwarg to an arg in all of the relevant methods.